src/algorithm/mod.rs

```diff
@@ -68,33 +68,50 @@ impl Algorithm {
                 &self.patterns
             ).is_empty()
         {
-            return Fscore::Win;
+            return Fscore::Value(10000000000 * depth as isize);
         }
         let three_cross_four = extract_missing_bit_cross_three_with_four(*player, *enemy);
         if three_cross_four.is_any() {
-            result += three_cross_four.count_ones() as isize * if node.is_players_last_move() { 500 } else { 1000 };
+            result += three_cross_four.count_ones() as isize * if !node.is_players_last_move() { 500 } else { 1000 };
             // result += three_cross_four.count_ones() as isize * 100;
         }
         let four_cross_four = extract_missing_bit_cross_four_with_four(*player, *enemy);
         if four_cross_four.is_any() {
-            result += four_cross_four.count_ones() as isize * if node.is_players_last_move() { 750 } else { 1200 };
+            result += four_cross_four.count_ones() as isize * if !node.is_players_last_move() { 750 } else { 1200 };
             // result += four_cross_four.count_ones() as isize * 200;
         }
         // TODO: Let this be a global static
-        let patterns: [((u8, u8, bool), isize); 10] = [
-            (self.patterns[PatternName::OpenThree], 500isize),
-            (self.patterns[PatternName::CloseThree], 50isize),
-            (self.patterns[PatternName::OpenSplitThreeLeft], 490isize),
-            (self.patterns[PatternName::OpenSplitThreeRight], 490isize),
-            (self.patterns[PatternName::OpenFour], 1100isize),
-            (self.patterns[PatternName::CloseFour], 1000isize),
-            (self.patterns[PatternName::SplitFourRight], 1000isize),
-            (self.patterns[PatternName::SplitFourLeft], 1000isize),
-            (self.patterns[PatternName::SplitFourMiddle], 1000isize),
-            (self.patterns[PatternName::Five], 5000isize)
+        // let patterns: [((u8, u8, bool), isize, isize); 12] = [
+        //     (self.patterns[PatternName::OpenThree], 50isize, 500isize),
+        //     (self.patterns[PatternName::CloseThree], 50isize, 1500isize),
+        //     (self.patterns[PatternName::OpenSplitThreeLeft], 50isize, 500isize),
+        //     (self.patterns[PatternName::OpenSplitThreeRight], 50isize, 500isize),
+        //     (self.patterns[PatternName::OpenFour], 9999999isize, 99999999isize),
+        //     (self.patterns[PatternName::CloseFour], 1000isize, 99999999isize),
+        //     (self.patterns[PatternName::SplitFourRight], 50isize, 99999999isize),
+        //     (self.patterns[PatternName::SplitFourLeft], 50isize, 99999999isize),
+        //     (self.patterns[PatternName::SplitFourMiddle], 50isize, 99999999isize),
+        //     (self.patterns[PatternName::Five], 10000000000isize, 100000000isize),
+        //     ((0b01100000, 4, true), 500isize, 2000isize),
+        //     ((0b01010000, 5, true), 25isize, 1000isize)
+        // ];
+        // Got score values from https://playgomoku.online/gomoku-offline
```

Comment:
Maybe not the greatest idea since it's a no depth game and might be optimized only to play no at 0 depth

Reply:
Could be true, but it actually seems to do a better job since I change those scores